nosnoc/nosnoc.py

```diff
@@ -401,7 +444,7 @@ def __init__(self,
             if (opts.irk_representation
                     in [IrkRepresentation.INTEGRAL, IrkRepresentation.DIFFERENTIAL_LIFT_X]):
                 self.add_variable(SX.sym(f'X_{ctrl_idx}_{fe_idx}_{ii+1}', dims.n_x), self.ind_x,
-                                  -inf * np.ones(dims.n_x), inf * np.ones(dims.n_x), model.x0, ii)
+                                  ocp.lbx, ocp.ubx, model.x0, ii)
```

Comment:
The matlab implementation has the ability to only apply the bounds on the control intervals, which is something we may want to do here as well.

Reply:
So  many options :sweat_smile: 
This one indeed doesn't seem too important.